source/game/kart/KartCollide.cc

```diff
@@ -438,6 +459,31 @@ void KartCollide::calcBoundingRadius() {
     m_boundingRadius = collisionGroup()->boundingRadius() * move()->hitboxScale();
 }
 
+/// @addr{0x80571F10}
+void KartCollide::calcObjectCollision() {
+    m_totalReactionWallNrm = EGG::Vector3f::zero;
+    m_surfaceFlags.resetBit(eSurfaceFlags::ObjectWall, eSurfaceFlags::ObjectWall3);
+
+    size_t collisionCount = objectCollisionKart()->checkCollision(pose(), velocity());
+
+    const auto *objectDirector = Field::ObjectDirector::Instance();
+
+    for (size_t i = 0; i < collisionCount; ++i) {
```

Comment:
```suggestion
    const auto *objectDirector = Field::ObjectDirector::Instance();

    for (size_t i = 0; i < objectCollisionKart()->checkCollision(pose(), velocity()); ++i) {
```

Reply:
These are not equivalent; in your suggestion, `checkCollision` will get called on each iteration of the loop. Is this intentional

Reply:
Yeah, you're right. I forgot about that.